### DISCUSSION TITLE
Duda sobre TPI3 - Eventos infinitos

### DISCUSSION THREAD

**Discussion by iancrespiok** [dds-jv/foro#278] [Consultas generales] [answered]
Hola! Tengo una duda sobre el pedido en el TPI de ver si un evento esta solapado con otro. Yo modele un Evento abstracto y de esa clase heredan EventoUnico y EventoRepetitivo (las cuales tienen un atributo LocalDate fechaLimite que puede ser null en caso que se infinito). 
En los EventoUnico es facil ya que hay una sola "iteracion" de dicho evento. Pero en los repetitivos es un problema.

Si me hacen por ejemplo:
eventoUnico.estaSolapadoCon(eventoRepetitivo) 
Se me ocurre que se puede hacer un for que vaya iterando fecha por fecha que sucede el evento repetitivo y compararla con la fecha del evento unico. Eso es viable, es finito. Puede tardar su tiempo pero sucedera. 
Idem para: eventoRepetitivo.estaSolapadoCon(eventoUnico) 

El problema es si me piden algo como:
eventoRepetitivo.estaSolapadoCon(eventoRepetitivo2)
Ahi lo unico que se me ocurre es generar dos List que contengan todas las repeticiones y ver si hay algun elemento LocalTimeDate que este en ambas List. 
Dos problemas, primero: es extremadamente largo el proceso e ineficiente y el segundo: solo sirve para EventosRepetitivos con fechaLimite. 

No se como solucionarlo. gracias

**Comment by ariel-** [comment]
Estaba buscando algo parecido para manejar las fechas repetidas sin almacenarlas, y se me ocurrió meterlas en un Stream!

Lamentablemente las API que mejor se adaptan están a partir de Java 9:
https://howtodoinjava.com/java9/stream-dates-datesuntil/

Justo esa devuelve un stream de fechas tomando una fecha limite y un período de repetición... me suena.

Se podrá, para este TP, utilizar Java 9?

**Comment by gastonprieto** [comment, marked as answer]
perdón por la demora @iancrespiok 

no nos va a importar ni la eficiencia, ni resolver el algoritmo que resuelva lo del infinito. En otra iteración se vera como se resuelve. El enfoque acá esta mas en las abstracciones que generan y donde estan las responsabilidades.
